repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala

```diff
@@ -188,11 +186,15 @@ class SamlPolicyTranslationFilter @Inject()(configurationService: ConfigurationS
     * @throws SamlPolicyException if parsing fails
     */
   def readToDom(samlResponse: InputStream): Document = {
+    var documentBuilder : DocumentBuilder = null
     try {
-      documentBuilderFactory.newDocumentBuilder().parse(samlResponse)
+      documentBuilder = XMLParserPool.borrowParser
+      documentBuilder.parse(samlResponse)
     } catch {
       case se: SAXException =>
         throw SamlPolicyException(SC_BAD_REQUEST, "SAMLResponse was not able to be parsed", se)
+    } finally {
+      if (documentBuilder != null) XMLParserPool.returnParser(documentBuilder)
```

Comment:
No curly on if, i'm surprised the style checker didn't catch that one. I went with an optional for the xpath version of this.